src/Ztm.WebApi/SqlCallbackRepository.cs

```diff
@@ -32,27 +37,27 @@ public async Task<Callback> AddAsync(IPAddress ip, DateTime requestTime, Uri url
                 throw new ArgumentNullException(nameof(url));
             }
 
-            var callback = new Callback(Guid.NewGuid(), ip, requestTime, false, url);
-
             using (var db = this.db.CreateDbContext())
-            using (var dbtx = await db.Database.BeginTransactionAsync(IsolationLevel.Serializable, cancellationToken))
             {
-                await db.WebApiCallbacks.AddAsync(ToEntity(callback));
-                await db.SaveChangesAsync(cancellationToken);
+                var callback = await db.WebApiCallbacks.AddAsync(new WebApiCallback()
+                {
+                    RegisteredIp = ip,
+                    RegisteredTime = DateTime.Now.ToUniversalTime(),
+                    Url = url,
+                }, cancellationToken);
 
-                var webApiCallback = await db.WebApiCallbacks.FirstOrDefaultAsync(c => c.Id == callback.Id);
-                dbtx.Commit();
+                await db.SaveChangesAsync(cancellationToken);
 
-                return ToDomain(webApiCallback);
+                return ToDomain(callback.Entity);
             }
         }
 
-        public async Task<Callback> SetStatusAsCompletedAsync(Guid id, CancellationToken cancellationToken)
+        public async Task SetCompletedAsyc(Guid id, CancellationToken cancellationToken)
         {
             using (var db = this.db.CreateDbContext())
             using (var dbtx = await db.Database.BeginTransactionAsync(IsolationLevel.Serializable, cancellationToken))
             {
-                var update = await db.WebApiCallbacks.FirstOrDefaultAsync(c => c.Id == id, cancellationToken);
+                var update = await db.WebApiCallbacks.FindAsync(new object[]{ id }, cancellationToken);
```

Comment:
Better to use previous version for more efficient due to the query is expression tree.